clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -3430,6 +3430,7 @@ Sema::InstantiateClass(SourceLocation PointOfInstantiation,
     llvm::raw_string_ostream OS(Name);
     Instantiation->getNameForDiagnostic(OS, getPrintingPolicy(),
                                         /*Qualified=*/true);
+    OS << ", file:" << SourceMgr.getFilename(Instantiation->getLocation());
```

Comment:
Would it be possible to add it in a structured manner rather than as a substring?
We are outputting JSON after all, it would be much more convenient to keep `args.detail` as is and add another one (e.g. `args.detail_file`)?

Reply:
Added it as a json parameter.